### PR TITLE
Pull out the docker client

### PR DESCRIPTION
**Pull request by rthallisey**
Take out the docker client creation from the apbs package and put it along side the kubernetes client.

Partially-fixes: #213 
Depends-on: #214

### CODE REVIEW

**Comment by rthallisey** [comment]
I'm trying to split up my commits since each of these will be easier to review individually and they are each a feature.

**Comment by rthallisey** [comment]
Testing: Run the normal broker tests.

File: pkg/apb/client.go [hunk 1]
```diff
@@ -40,8 +39,6 @@ First cut might have to pass kubecfg from broker. FIRST SPRINT broker passes use
 admin/admin
 */
 
-var DockerSocket = "unix:///var/run/docker.sock"
```
Comment (jmrodri) [inline pkg/apb/client.go]: good

File: pkg/apb/client.go [hunk 2]
```diff
@@ -56,52 +53,23 @@ type Client struct {
...
-	config, err := clientcmd.NewDefaultClientConfig(*clientConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
-	if err != nil {
-		return nil, err
-	}
-	return config, nil
-}
-
 func NewClient(log *logging.Logger) (*Client, error) {
-	dockerClient, err := docker.NewClient(DockerSocket)
+	dockerClient, err := clients.Docker(log)
```
Comment (jmrodri) [inline pkg/apb/client.go]: ok that's cleaner

File: pkg/clients/docker.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package clients
+
+import (
+	docker "github.com/fsouza/go-dockerclient"
+	logging "github.com/op/go-logging"
+)
+
+func Docker(log *logging.Logger) (*docker.Client, error) {
+	dockerClient, err := docker.NewClient(DockerSocket)
```
Comment (jmrodri) [inline pkg/clients/docker.go]: This is nicer.

File: pkg/clients/kubernetes.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	clientConfig, err := clientcmd.LoadFromFile(configPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := clientcmd.NewDefaultClientConfig(*clientConfig, &clientcmd.ConfigOverrides{}).ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
```
Comment (jmrodri) [inline pkg/clients/kubernetes.go]: :bulb: Ok so this file is from the previous PR. Finally understood what's going on. I find the dependent PRs much harder to review. I would've preferred a clients feature to do all of the cleanup at once.

File: pkg/clients/types.go [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package clients
+
+const (
+	DockerSocket = "unix:///var/run/docker.sock"
+)
```
Comment (jmrodri) [inline pkg/clients/types.go]: Great! :+1:

**Comment by jmrodri** [comment]
Approving this PR as well. Again removing `Clients struct` and reverting `EtcdConfig` to `Config`.

File: pkg/clients/docker.go [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package clients
+
+import (
+	docker "github.com/fsouza/go-dockerclient"
+	logging "github.com/op/go-logging"
+)
+
+func Docker(log *logging.Logger) (*docker.Client, error) {
```
Comment (shawn-hurley) [inline pkg/clients/docker.go]: Before Merge, I think this needs doc comments.

**Comment by rthallisey** [comment]
Working in #222

**Pull request closed without merging**